mdoc/src/main/scala-3/mdoc/internal/markdown/MarkdownCompiler.scala

```diff
@@ -194,7 +222,8 @@ class MarkdownCompiler(
           } else {
             msg
           }
-        reportMessage(vreporter, diagnostic, mpos, actualMessage)
+
+        reportMessage(vreporter, diagnostic, mpos, "\n" + actualMessage)
```

Comment:
@tgodzik this might interest you as it affects worksheets. To give similar experience in markdown document, I had to prepend a newline to dotty's messages to make it look similar to what scala 2 does:

```

    """|error: yrangepos-scala3.md:3:11:
       |Found:    String
       |Required: Int
       |List[Int]("".length.toString)
       |          ^^^^^^^^^^^^^^^^^^
```

What actually comes from the compiler is `Found:    String\nRequired: Int`, unlike scala 2, which does `type mismatch:\nfound:...`

Will this new line be problematic for rendering in worksheets? (I think some of the worksheet tests got broken by this)

Reply:
It should not be problematic as it's rendered in the diagnostics of the editors. I am not sure I understand the issue here. How would it look like without the additional `\n` ?

Reply:
Without the newline it'd look like this:

```
error: yrangepos-scala3.md:3:11:Found:    String
Required: Int
List[Int]("".length.toString)
          ^^^^^^^^^^^^^^^^^^ 
```

Whereas for scala 2 it looks like this:

```
error: yrangepos.md:3:11: type mismatch;
 found   : String
 required: Int
List[Int]("".length.toString)
          ^^^^^^^^^^^^^^^^^^
```

So I added the newline for consistency.

And it's the same for all the other errors from dotty. I can remove the newline, but I think it will look worse in markdown than scala 2 version :)

Reply:
I think it's fine then. Especially if it worked like that in Scala 2, then we already handled it correctly in worksheets. Anyway, thanks for letting me know!

Reply:
Cool!

There are some strange errors remaining with regards to Scala 3 new syntax and mdoc generated code, but I think I'm on good track to finish all the tests this week.

If needed I will separate the PR (I started fixing tests to verify my changes to MarkdownCompiler were okay) as outlined in the original plan.

Reply:
Awesome, let me know when we should take a look at it!